kubernetes/loculus/silo_import_job.sh

```diff
@@ -148,6 +151,46 @@ download_data() {
   echo
 }
 
+# Generate the preprocessing config file with the lineage file for the current pipeline version.
+# the lineage definition file needs to be downloaded first.
+prepare_preprocessing_config() {
+  rm -f $lineage_definition_file $preprocessing_config_file_merged
+
+  if [[ -z "$LINEAGE_DEFINITIONS" ]]; then
+    echo "No LINEAGE_DEFINITIONS given, nothing to configure;"
+    cp $preprocessing_config_file $preprocessing_config_file_merged
+    return
+  fi
+
+  pipelineVersion=$(zstd -d -c "$new_input_data_path" | jq -r '.metadata.pipelineVersion' | sort -u)
+
+  if [[ -z "$pipelineVersion" ]]; then
+    echo "No pipeline version found. Writing empty lineage definition file."
+    touch $lineage_definition_file
+  elif [[ $(echo "$pipelineVersion" | wc -l) -eq 1 ]]; then
+    echo "Single pipeline version: $pipelineVersion"
+
+    # Get the URL for the version from LINEAGE_DEFINITIONS
+    lineage_url=$(echo "$LINEAGE_DEFINITIONS" | jq -r --arg version "$pipelineVersion" '.[$version]')
+    if [[ -z "$lineage_url" || "$lineage_url" == "null" ]]; then
+      echo "Error: No URL defined for pipeline version $pipelineVersion."
+      exit 1
+    fi
+
+    # Download the file from the URL
+    if ! curl -s -o "$lineage_definition_file" "$lineage_url"; then
+      echo "Error: Failed to download file from $lineage_url."
+      exit 1
+    fi  
+  else
+    echo "Multiple pipeline versions in data to import: $pipelineVersion"
+    exit 1
+  fi
+
+  cp $preprocessing_config_file $preprocessing_config_file_merged
+  echo -e "lineageDefinitionsFilename: \"$lineage_definition_file\"\n" >> $preprocessing_config_file_merged
```

Comment:
Does it make sense to wait for the new SILO version that supports supplying this via command line args? I don't know how urgent this change is. https://github.com/GenSpectrum/LAPIS-SILO/pull/633 shouldn't take too long anymore.

Reply:
I don't think this PR is urgent in principle, I think we can wait a bit more, but I would probably want to get this PR closed this week, so I can close everything before the holidays. In that case I'd rather make an issue and link the PR so we can change it later. It's not really that bad the way the config is assembled now I think.